docs/dev/framework/caching.md

```diff
@@ -244,24 +244,104 @@ Moreover, you don't have to register to all the different callbacks such as `ons
-with a `Cache-Control: private` header for example, then the page on which the fragment
-is visible cannot be cached. On the other hand, if the fragment can be cached, it
-can provide its own cache tags, as mentioned previously.
-
-Contao brings its own `forward` fragment renderer, which provides the fragment with 
-a full clone of the request. This provides the fragment with the full `POST` data
-for example, contrary to Symfony's default `inline` renderer.
-
-The renderer can also be set to `esi`. In that case, if Symfony detects that it 
-is talking to  a gateway cache that supports ESI (like Symfony's built in reverse 
-proxy, that Contao uses), it generates an ESI include tag. See also [Symfony's documentation][esi] 
-on _Edge Side Includes_.
+_fragment controllers_. These fragments are rendered with their defined renderer
+and merged into the main content. Each fragment returns a response and thus can
+tell whether it can be cached or not.
+
+
+There are two fundamentally different ways to render fragments:
+ 1. **Inline:** Inside the main page request and merged with the content before caching. In Symfony, this is handled
+    by the `inline` renderer, but Contao provides its own default `forward` renderer, which copies the original
+    request (whereas Symfony does not) to e.g. pass POST data to the subrequest.
+ 2. **Subrequest:** By generating a placeholder for the fragment, the fragment content is rendered **after** caching
+    the page, which allows a fragment to have a different cache time than the main page. In case of `esi` the reverse
+    proxy (cache) will parse these placeholders before delivering to the client.
+    There's also an `hinclude` renderer which allows browsers to merge fragments using JavaScript, but the client-side
+    part is not provided out-of-the-box.
+
+
+### Inline Fragments
+
+{{% notice note %}}
+Before **Contao 4.9**, inline fragments cannot affect the cache time of the page response.
+{{% /notice %}}
+
+By default, fragments like frontend modules and content elements are rendered directly inside the main request.
+This means the content of an inline fragment will be cached for as long as the page is cached, as configured
+in the page setting. If a fragment returns caching information in its response, the page's cache time and the fragment
+will be merged to the lowest common denominator. For example, if the page is cacheable for a day, but the fragment
+only for one hour, the whole page will only be cached for one hour.
+
+A more common use case would be something like a `{{date::Y}}` insert tag, which renders the current year. This
+fragment is cacheable until the end of 31st of December. If the page is cached on the 10th of December for one day,
+the page cache time will not be affected. However, if the page is cached at 12pm on the 31st of December, the cache
+time will be lowered to 12 instead of 24 hours. The page cache will expire at the end of the year and regenerated
+on the 1st of January.
```

Comment:
How about this instead:

On the other hand, consider an example of a fragment that renders the current year. This fragment would be cacheable
until the end of 31st of December. If the page is cached on the 10th of December for one day, the page cache time will
not be affected. However, if the page is cached at 12pm on the 31st of December, the cache time will be lowered to 12
instead of 24 hours. The page cache will expire at the end of the year and regenerated on the 1st of January.

```php
class CurrentYearController extends AbstractFrontendModuleController
{
    protected function getResponse(Template $template, ModuleModel $model, Request $request): ?Response
    {
        $year = (int) date('Y');
        $template->year = $year;
        $response = $template->getResponse();

        // Cache until the end of the year
        $response->setPublic();
        $response->setMaxAge(strtotime($year.'-12-31 23:59:59') - time());
        
        return $response;
    }
}
```

Reply:
Ping @Toflar @aschempp 

If you agree with this change, I can add it directly to Andreas' PR and then merge this. I'd like to finally get this along, since the PR is almost a year old now 😬

Reply:
Yeah go ahead. We better merge that change. I'll go over the whole article again once it's live and I can see it in full glory ;) just ping me on slack and I'll review again.

Reply:
Alright :)